features/ui/button/button.tsx

```diff
@@ -1,9 +1,45 @@
-import styled from "styled-components";
+import React, { ReactText, ReactElement, ReactNode } from "react";
+import styled, { css } from "styled-components";
+import { color, textFont, space } from "@styles/theme";
 
+export enum ButtonSize {
+  sm = "sm",
+  md = "md",
+  lg = "lg",
+  xl = "xl",
+}
+
+export enum ButtonColor {
+  primary = "primary",
+  secondary = "secondary",
+  gray = "gray",
+  empty = "empty",
+  emptyGray = "emptyGray",
+  error = "error",
+}
+
+export enum ButtonIcon {
+  leading = "leading",
+  trailing = "trailing",
+  only = "only",
+  none = "none",
+}
+
+export const Icon = styled.img<{ icon: ButtonIcon }>`
+  display: block;
+  width: ${space(5)};
+  margin-left: ${(props) =>
+    props.icon == ButtonIcon.trailing ? ".5rem" : "none"};
+  margin-right: ${(props) =>
+    props.icon == ButtonIcon.leading ? ".5rem" : "none"};
+`;
+
+// This below Button variable is only here because there's at least one
+// instance where it is being used throughout the app
```

Comment:
I think it would make sense to remove this Button and replace it with the new version. You can either use these styles directly in the navigation (where it is currently used) or try to use the new button component there as well. I tried that and it is actually more challenging than expected 😄